ruby/ql/lib/codeql/ruby/frameworks/http_clients/Typhoeus.qll

```diff
@@ -55,3 +68,30 @@ private module TyphoeusDisablesCertificateValidationConfig implements DataFlow::
 
 private module TyphoeusDisablesCertificateValidationFlow =
   DataFlow::Global<TyphoeusDisablesCertificateValidationConfig>;
+
+/** Gets the body from the given `requestNode` representing a Typhoeus request */
+private DataFlow::Node getBodyFromRequest(API::Node requestNode) {
```

Comment:
This predicate and the ones below are likely to be the cause of the performance issue.

See [this doc comment](https://github.com/github/codeql/blob/fd82a06b93a850b2f4bb9594933574762d8981d7/ruby/ql/lib/codeql/ruby/ApiGraphs.qll#L126) about strict left-to-right evaluation and the [associated footgun](https://github.com/github/codeql/blob/fd82a06b93a850b2f4bb9594933574762d8981d7/ruby/ql/lib/codeql/ruby/ApiGraphs.qll#L137).

It's possible you can avoid this by adding the modifiers in front of the predicates:
```
bindingset[requestNode]
pragma[inline_late]
```

Otherwise restrict `requestNode` to the nodes you know to be sources of `Request` objects.